pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitTestsShouldIncludeAssert.xml

```diff
@@ -429,6 +429,23 @@ class Style {
   public void moveOutOfBoundsFrom() {
     doSomething();
   }
+}]]></code>
+    </test-code>
+    <test-code>
+        <description>#??? Treat AssertJ soft assertions as assert expressions</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+class FooTest {
+
+    @Test
+    void testFoo() {
+        var softly = new SoftAssertions();
+        softly.assertThat("doesn't matter").isEqualTo("doesn't matter");
+        softly.assertAll();
+    }
 }]]></code>
     </test-code>
```

Comment:
Could you please also add a test case with a junit Rule? e.g.

```java
class FooTest {
    @Rule
    public JUnitSoftAssertions softAssertions = new JUnitSoftAssertions();

    @Test
    void testFoo() {
        softAssertions.assertThat(1).isEqualTo(1);
        softAssertions.assertAll();
    }
}
```
(see https://blog.codeleak.pl/2015/09/assertjs-softassertions-do-we-need-them.html)

For this to work, we probably need to change the rule to look for subtypes of `org.assertj.core.api.AbstractSoftAssertions` instead of `org.assertj.core.api.SoftAssertions`.

Reply:
Except it is a little more complex, as the `JUnitSoftAssertions` rule does not need have a call to `assertAll` done in test methods. I will give it a try.